### ISSUE TITLE
Rule proposal: `prefer-dataset`

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#166]
[<img alt="Issuehunt badges" src="https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/166/badge.svg" />](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/166)

Prefer the [`.dataset`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) property on DOM elements over using `setAttribute`.

We would detect `setAttribute` and that the string starts with `data-`.

## Fail

```js
element.setAttribute('data-unicorn', '🦄');
```

## Pass

```js
element.dataset.unicorn = '🦄';
```

> There is a $60.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/166).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

**Comment by lexcorp16** [comment]
@sindresorhus how can I get started with this?

**Comment by sindresorhus** [comment]
@lexcorp16 https://github.com/sindresorhus/eslint-plugin-unicorn/blob/d083f162a2ccaeb730e4ae284ac0af4d7c04d01f/docs/new-rule.md :)

**Comment by udaypydi** [comment]
if @lexcorp16  has not started, I would like to get started with this @sindresorhus .

**Comment by IssuehuntBot** [comment]
@issuehuntfest has funded $60.00 to this issue. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/166)

**Comment by EROPLM** [comment]
@sindresorhus Where is the file writing this code?
